packages/flutter/test/material/snack_bar_test.dart

```diff
@@ -2280,6 +2280,61 @@ void main() {
       ),
     );
   });
+
+  testWidgets('SnackBarAction backgroundColor works as expected', (WidgetTester tester) async {
+    final MaterialStateColor backgroundColor = MaterialStateColor.resolveWith((Set<MaterialState> states) {
+      if (states.contains(MaterialState.disabled)) {
+        return Colors.blue;
+      }
+      return Colors.purple;
+    });
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: Builder(
+            builder: (BuildContext context) {
+              return GestureDetector(
+                onTap: () {
+                  ScaffoldMessenger.of(context).showSnackBar(
+                    SnackBar(
+                      content: const Text('I am a snack bar.'),
+                      duration: const Duration(seconds: 2),
+                      action: SnackBarAction(
+                        backgroundColor: backgroundColor,
+                        label: 'ACTION',
+                        onPressed: () {},
+                      ),
+                    ),
+                  );
+                },
+                child: const Text('Tap'),
+              );
+            },
+          ),
+        ),
+      ),
+    );
+
+    await tester.tap(find.text('Tap'));
+    await tester.pump();
+    await tester.pump(const Duration(milliseconds: 750));
```

Comment:
Does a `tester.pumpAndSettle` work instead of a hardcoded duration? If not, could you add a comment describing why we're using the hardcoded duration here instead of `pumpAndSettle`?

Reply:
This is a good point. Unless the duration has a rationale, in other words unless we're waiting 750ms because we understand exactly what's going to occupy those 750ms, it's best to use `pumpAndSettle`.